joblib/_dask.py

```diff
@@ -286,12 +286,19 @@ async def maybe_to_futures(args):
                             # calling client.scatter inside a dask worker)
                             # using hash=True often raise CancelledError,
                             # see dask/distributed#3703
-                            [f] = await self.client.scatter(
+                            _coro = self.client.scatter(
                                 [arg],
                                 asynchronous=True,
                                 hash=False
                             )
-                            call_data_futures[arg] = f
+                            # Centralize the scattering of identical arguments
+                            # between concurrent apply_async callbacks by
+                            # exposing the running coroutine in
+                            # call_data_futures before it completes.
+                            t = asyncio.Task(_coro)
+                            call_data_futures[arg] = t
+
+                            f = await t
```

Comment:
Maybe this is the cause of the problem?

```suggestion
                            [f] = await t
```

Reply:
actually we could just avoid using `[arg]` (and use `arg`) instead in the scatter call.

Reply:
we probably need a test for this, because the existing tests did not fail :)